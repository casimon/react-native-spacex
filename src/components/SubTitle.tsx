import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle
} from 'react-native';

const styles = StyleSheet.create({
  title: {
    paddingHorizontal: 6,
    paddingVertical: 3
  },
  titleText: {
    fontSize: 24,
    color: '#000000'
  }
});

interface Props {
  text: string;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
}

class SubTitle extends React.PureComponent<Props> {
  render() {
    return (
      <View style={[styles.title, this.props.style]}>
        <Text
          style={[styles.titleText, this.props.styleText]}
          allowFontScaling={false}>
          {this.props.text}
        </Text>
      </View>
    );
  }
}

export default SubTitle;
