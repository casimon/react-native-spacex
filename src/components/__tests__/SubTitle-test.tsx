/**
 * @format
 */

import 'react-native';
import React from 'react';
import SubTitle from '../SubTitle';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders sub title with text', () => {
  const tree = renderer.create(<SubTitle text="Test text prop" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders sub title with text and optional style prop', () => {
  const tree = renderer
    .create(<SubTitle text="Test text" style={{margin: 2}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders sub title with text and optional styleText prop', () => {
  const tree = renderer
    .create(<SubTitle text="Test text" styleText={{fontSize: 12}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
