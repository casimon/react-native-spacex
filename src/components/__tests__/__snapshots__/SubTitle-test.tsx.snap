// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sub title with text 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
        },
        undefined,
      ]
    }
  >
    Test text prop
  </Text>
</View>
`;

exports[`renders sub title with text and optional style prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      Object {
        "margin": 2,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
        },
        undefined,
      ]
    }
  >
    Test text
  </Text>
</View>
`;

exports[`renders sub title with text and optional styleText prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
        },
        Object {
          "fontSize": 12,
        },
      ]
    }
  >
    Test text
  </Text>
</View>
`;
